server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java

```diff
@@ -382,6 +372,55 @@ public TSStatus deleteRegion(TConsensusGroupId tconsensusGroupId) throws TExcept
     }
   }
 
+  @Override
+  public TMigrateRegionResp migrateRegion(TMigrateRegionReq req) throws TException {
+    TRegionReplicaSet regionReplicaSet = req.migrateRegion;
+    TSStatus tsStatus;
+    ConsensusGenericResponse consensusGenericResponse;
+    switch (regionReplicaSet.regionId.type) {
+      case DataRegion:
+        DataRegionId dataRegionId = new DataRegionId(regionReplicaSet.getRegionId().getId());
+        List<Peer> newPeers = new ArrayList<>();
+        for (TDataNodeLocation dataNodeLocation : regionReplicaSet.getDataNodeLocations()) {
+          TEndPoint endpoint =
+              new TEndPoint(
+                  dataNodeLocation.getDataRegionConsensusEndPoint().getIp(),
+                  dataNodeLocation.getDataRegionConsensusEndPoint().getPort());
+          newPeers.add(new Peer(dataRegionId, endpoint));
+        }
+        consensusGenericResponse =
+            DataRegionConsensusImpl.getInstance().changePeer(dataRegionId, newPeers);
+        break;
+      case SchemaRegion:
+        SchemaRegionId schemaRegionId = new SchemaRegionId(regionReplicaSet.getRegionId().getId());
+        newPeers = new ArrayList<>();
+        for (TDataNodeLocation dataNodeLocation : regionReplicaSet.getDataNodeLocations()) {
+          TEndPoint endpoint =
+              new TEndPoint(
+                  dataNodeLocation.getSchemaRegionConsensusEndPoint().getIp(),
+                  dataNodeLocation.getSchemaRegionConsensusEndPoint().getPort());
+          newPeers.add(new Peer(schemaRegionId, endpoint));
+        }
+        consensusGenericResponse =
+            SchemaRegionConsensusImpl.getInstance().changePeer(schemaRegionId, newPeers);
+        break;
+      default:
+        // unsupported region type
+        tsStatus = new TSStatus(TSStatusCode.INTERNAL_SERVER_ERROR.getStatusCode());
+        tsStatus.setMessage("Region type is invalid");
+        return new TMigrateRegionResp(tsStatus);
+    }
+
+    if (consensusGenericResponse.isSuccess()) {
+      tsStatus = new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+    } else {
+      tsStatus = new TSStatus(TSStatusCode.INTERNAL_SERVER_ERROR.getStatusCode());
```

Comment:
We should avoid using the INTERNAL_SERVER_ERROR. Otherwise, the interface user would never know what's wrong with the internal server.